Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -997,16 +996,15 @@ - (void)applyReferralCode:(NSString *)code andCallback:(callbackWithParams)callb
 
 #pragma mark - Private methods
 
-+ (Branch *)getBranchInstance:(BOOL)isLive {
-    if (!currInstance) {
-        NSString *branchKey = [BNCPreferenceHelper getBranchKey:isLive];
-        if (!branchKey || [branchKey isEqualToString:NO_STRING_VALUE]) {
-            NSLog(@"Branch Warning: Please enter your branch_key in the plist!");
-        }
-        
-        [Branch initInstance];
-    }
-    return currInstance;
++ (Branch *)getInstanceInternal {
+    static Branch *branch;
```

Comment:
@aaustin A static variable like this will not be cleaned up, Check out [this question and answer on SO](http://stackoverflow.com/a/15924867/1252541), and feel free to test it out yourself. This is how I create most of my singletons, so there can't be any accidental direct references elsewhere in the code.